src/main/java/org/eclipse/uprotocol/streamer/UStreamer.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.streamer;
+
+import org.eclipse.uprotocol.v1.UCode;
+import org.eclipse.uprotocol.v1.UMessage;
+import org.eclipse.uprotocol.v1.UStatus;
+import org.eclipse.uprotocol.v1.UUri;
+import org.eclipse.uprotocol.transport.UListener;
+
+import org.eclipse.uprotocol.core.usubscription.v3.FetchSubscribersRequest;
+import org.eclipse.uprotocol.core.usubscription.v3.FetchSubscriptionsRequest;
+import org.eclipse.uprotocol.core.usubscription.v3.FetchSubscriptionsResponse;
+import org.eclipse.uprotocol.core.usubscription.v3.NotificationsRequest;
+import org.eclipse.uprotocol.core.usubscription.v3.SubscriberInfo;
+import org.eclipse.uprotocol.core.usubscription.v3.USubscription;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/** 
+ * Example implementation of "Pluggable Streamer" written in Java.
+ * 
+ * The streamer provides APIs to simply add remove forwarding rules. A rule is defined as a 
+ * combination of input and output routes. The input route is the route to listen to and the output route is the route to forward the message to.
+ * The forwarding works by registering a listener on the input route and forwarding the message to the output route.
+ */
+public class UStreamer {
+
+    // List of listeners (routes) that the streamer listens to
+    private List<TransportListener> listeners;
```

Comment:
what is the advantage of the TransportListener?
why can't we just have a list of Route objects?

Reply:
The routes are not the rules in the forwarding table, the rules are a combination of in and out Route that then build a TransportListener and all we need to keep track of is the TransportListeners.